app.py

```diff
@@ -1,4 +1,4 @@
-from flask import Flask, redirect, render_template
+from flask import Flask, redirect, render_template, request, abort
```

Comment:
It seems there is a slight conflict between the import of request and abort, as the main branch does not import request yet. After that is fixed, it will be good to merge as the code looks great